datastore/gcloud/aio/datastore/array.py

```diff
@@ -0,0 +1,43 @@
+try:
+    from collections.abc import Sequence
+except ImportError:
+    from collections import Sequence
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import List
+
+import gcloud.aio.datastore.value
+
+
+# https://cloud.google.com/datastore/docs/reference/data/rest/v1/projects/runQuery#ArrayValue
+class Array(Sequence):
+
+    def __init__(self, items: List[Any]) -> None:
+        super(Array, self).__init__()
+        self.items = items
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Array):
+            return False
+        return self.items == other.items
+
+    def __repr__(self) -> str:
+        return repr(self.items)
+
+    @classmethod
+    def from_repr(cls, data: Dict[str, Any]) -> 'Array':
+        return cls([gcloud.aio.datastore.Value.from_repr(x)
+                    for x in data['values']])
+
+    def to_repr(self) -> Dict[str, Any]:
+        return {'values': [x.to_repr() for x in self]}
+
+    def __iter__(self) -> Iterator:
```

Comment:
Do you need to define this? I believe the `Sequence` class will already do this for you (since you've defined `__getitem__`) -- see [the parent implementation](https://github.com/python/cpython/blob/3.8/Lib/_collections_abc.py#L870)